dotCMS/src/main/java/com/dotcms/vanityurl/model/CacheVanityKey.java

```diff
@@ -0,0 +1,24 @@
+package com.dotcms.vanityurl.model;
+
+import com.dotcms.repackage.org.apache.commons.collections.keyvalue.MultiKey;
+
+import java.io.Serializable;
+
+/**
+ * Encapsulates the keys for cache vanities.
+ * @author jsanca
+ */
+public class CacheVanityKey extends MultiKey implements Serializable {
+
+    public CacheVanityKey(final String hostId, final long languageId, final String uri) {
+        super(hostId, languageId, uri);
+    }
+
+    @Override
+    public String toString() {
+
+        return this.getKey(0) + "|" +
+                this.getKey(2).toString().replace('/', '|') +
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Assign this magic number 2 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS109)